src/overlays/kaleido_scope/ovl_kaleido_scope/z_kaleido_scope_NES.c

```diff
@@ -502,8 +516,965 @@ s32 D_8082BEB8[] = { 0, 0 };
+
+            pauseCtx->nameSegment = (void*)ALIGN16((uintptr_t)pauseCtx->iconItemLangSegment + size2);
+            func_8011552C(play, 0x15);
+            if (((void)0, gSaveContext.worldMapArea) < 0x16) {
+                func_8082192C(pauseCtx->nameSegment + 0x400, ((void)0, gSaveContext.worldMapArea));
+            }
+
+            pauseCtx->iconItemVtxSegment = (void*)ALIGN16((uintptr_t)pauseCtx->nameSegment + 0xA00);
+            DmaMgr_SendRequest0(pauseCtx->iconItemVtxSegment, SEGMENT_ROM_START(icon_item_vtx_static),
+                                SEGMENT_ROM_SIZE(icon_item_vtx_static));
+
+            pauseCtx->unk_2B6 = 255;
+            pauseCtx->unk_2B7 = 255;
+            pauseCtx->unk_2B8 = 255;
+            pauseCtx->state = PAUSE_STATE_OPENING_3;
+            break;
+
+        case PAUSE_STATE_OPENING_3:
+            pauseCtx->infoPanelOffsetY += 10;
+            pauseCtx->itemPageRoll = pauseCtx->MapPageRoll = pauseCtx->QuestPageRoll = pauseCtx->MaskPageRoll -= 40.0f;
+
+            interfaceCtx->startAlpha += 63;
+            sPauseCursorLeftX += (s16)(sPauseCursorLeftMoveOffsetX / 4);
+            sPauseCursorRightX += (s16)(sPauseCursorRightMoveOffsetX / 4);
+            pauseCtx->alpha += 31;
+            if (pauseCtx->itemPageRoll == 0) {
+                interfaceCtx->startAlpha = 255;
+                sPauseMenuVerticalOffset = 0.0f;
+                pauseCtx->state = PAUSE_STATE_OPENING_4;
+            }
+
+            KaleidoScope_UpdateOpening(play);
+            break;
+
+        case PAUSE_STATE_OPENING_4:
+            pauseCtx->alpha += 31;
+            KaleidoScope_UpdateOpening(play);
+
+            if (pauseCtx->state == PAUSE_STATE_MAIN) {
+                KaleidoScope_UpdateNamePanel(play);
+            }
+            break;
+
+        case PAUSE_STATE_MAIN:
+            switch (pauseCtx->mainState) {
+                case PAUSE_MAIN_STATE_IDLE:
+                    if (!pauseCtx->itemDescriptionOn &&
+                        (CHECK_BTN_ALL(input->press.button, BTN_START) || CHECK_BTN_ALL(input->press.button, BTN_B))) {
+                        func_8011552C(play, 0xA);
+                        pauseCtx->state = PAUSE_STATE_UNPAUSE_SETUP;
+                        sPauseMenuVerticalOffset = -6240.0f;
+                        func_801A3AEC(0);
+                    }
+                    break;
+
+                case PAUSE_MAIN_STATE_SWITCHING_PAGE:
+                    KaleidoScope_UpdateSwitchPage(play, input);
+                    break;
+
+                case PAUSE_MAIN_STATE_SONG_PLAYBACK:
+                    // Song is in playback
+                    pauseCtx->ocarinaStaff = AudioOcarina_GetPlaybackStaff();
+                    if (pauseCtx->ocarinaStaff->state == 0) {
+                        // Song playback is finished, give song prompt to player
+                        pauseCtx->mainState = PAUSE_MAIN_STATE_SONG_PLAYER_PLAYING_INIT;
```

Comment:
I like "song prompt", we could use that for the ocarina staff system, instead of playback/playing (whichever uses player input, I always mix them up)

Reply:
Yeah, playing/playback is a bit similar at the moment:
`playing` -> `await user input`
`playback` -> `user is listening the game play back the song for them`

`playing` -> `prompt` (with throwing song in there) is an idea worth exploring


Side note, I have been following your docs on Kaleido in your OoT draft (and obv. where I got a lot of these names from).  Was going to comment on your draft soon, but thoughts on `mainState` vs .`mainSubState`?